core/crypto/poseidon_hash_test.go

```diff
@@ -0,0 +1,26 @@
+package crypto_test
+
+import (
+	"testing"
+
+	"github.com/NethermindEth/juno/core/crypto"
+	"github.com/NethermindEth/juno/core/felt"
+)
+
+func BenchmarkPoseidon(b *testing.B) {
+	e0, err := new(felt.Felt).SetString("0x3d937c035c878245caf64531a5756109c53068da139362728feb561405371cb")
+	if err != nil {
+		b.Errorf("Error occured %s", err)
+	}
+
+	e1, err := new(felt.Felt).SetString("0x208a0a10250e382e1e4bbe2880906c2791bf6275695e02fbbc6aeff9cd8b31a")
+	if err != nil {
+		b.Errorf("Error occured %s", err)
+	}
+
+	var f *felt.Felt
```

Comment:
Please refer to the section "A note on compiler optimisation" [here](https://dave.cheney.net/2013/06/30/how-to-write-benchmarks-in-go) for why the variable `f` should be declared as a package-level variable.